lib/media/ver10/osd.ex

```diff
@@ -0,0 +1,225 @@
+defmodule Onvif.Media.Ver10.OSD do
+  @moduledoc """
+  OSD (On-Screen Display) specification.
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+  import SweetXml
+
+  @primary_key false
+  @derive Jason.Encoder
+  @required [:token, :video_source_configuration_token, :type]
+  @optional []
+
+  embedded_schema do
+    field(:token, :string)
+    field(:video_source_configuration_token, :string)
+    field(:type, Ecto.Enum, values: [text: "Text", image: "Image", extended: "Extended"])
+
+    embeds_one :position, Position, primary_key: false, on_replace: :update do
+      @derive Jason.Encoder
+      field(:type, Ecto.Enum,
+        values: [
+          upper_left: "UpperLeft",
+          upper_right: "UpperRight",
+          lower_left: "LowerLeft",
+          lower_right: "LowerRight",
+          custom: "Custom"
+        ]
+      )
+
+      field(:pos, :map)
+    end
+
+    embeds_one :text_string, TextString, primary_key: false, on_replace: :update do
+      @derive Jason.Encoder
+      field(:is_persistent_text, :boolean)
+
+      field(:type, Ecto.Enum,
+        values: [plain: "Plain", date: "Date", time: "Time", date_and_time: "DateAndTime"]
+      )
+
+      field(:date_format, Ecto.Enum,
+        values: [
+          "M/d/yyyy": "M/d/yyyy",
+          "MM/dd/yyyy": "MM/dd/yyyy",
+          "dd/MM/yyyy": "dd/MM/yyyy",
+          "yyyy/MM/dd": "yyyy/MM/dd",
+          "yyyy-MM-dd": "yyyy-MM-dd",
+          "dddd, MMMM dd, yyyy ": "dddd, MMMM dd, yyyy ",
+          "MMMM dd, yyyy ": "MMMM dd, yyyy ",
+          "dd MMMM, yyyy": "dd MMMM, yyyy"
```

Comment:
this looks a little odd to me, I don't think we want to have string to atom conversion here. Instead maybe you can do a validator https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_inclusion/4 with the expected values in a list.

Reply:
Same for time format